### ISSUE TITLE
automate updates of the diagnostic messages page

### ISSUE THREAD

**Issue by kwalrath** [dart-lang/site-www#1949]
See #1892 for details

**Comment by kwalrath** [comment]
Here's the process I'm doing by hand:

1. Create a new branch of my site-www repo.
2. Go to https://github.com/dart-lang/sdk/blob/master/pkg/analyzer/tool/diagnostics/diagnostics.md, click the **Raw** button, and **Save as** _pathToRepo_/src/tools/diagnostic-messages.md.
3. View the diffs, and make sure they look reasonable.
4. Inform @bwilkerson of any issues. E.g., `title: Diagnostics` should be `title: Diagnostic messages`. Maybe edit the file if I don't feel like waiting.
5. Once the file is ready, create a PR.

**Comment by kwalrath** [comment]
Thinking about how to automate this, I'm not positive it should be automated, but perhaps it'd be good to have a build check (failable) about whether this file should be updated.

**Comment by chalin** [comment]
Yes, that is what I had in mind; similar to what we do with the code-excerpt refresh. I.e., actual code refresh is under site-maintainer control (and done manually using a single command), but the build scripts ensure that everything is up-to-date.

**Comment by parlough** [comment]
@devoncarew I'm not sure if you're the right person to ask, but I know you've been working with automation a bit recently. Let me know if I should direct my question elsewhere.

We're expecting the source file to change, perhaps to one or more YAML files, but we'd still like to automatic moving of 1+ diagnostic related files from the SDK to this repository in our work to better standardizing linter rule documentation as well as diagnostics.

I was thinking of using a GitHub workflow which checks for updates once a day, then if there are differences, open a new branch and pull request with the changes. Then use the existing autosubmit logic to automatically merge in the changes if all tests pass.

Does this make sense? If so, does Google have any rules for GitHub bots, particularly in relation to the CLA? Can I use the default github-actions bot, should I use one that the Dart team already has, work with the devrel team to set up a new one? 

Thanks so much for any thoughts you have!

**Comment by parlough** [comment]
Thanks for your thoughts!

I'd like to avoid the check that fails builds, since we've had a few in the past and they slow down writers.

The idea of shipping it alongside the SDK may be interesting and I'll consider that as we figure out what the files exactly look like and include. Another similar options is including the SDK as a submodule, that I've been mostly avoiding, but the PRs come easily with dependabot and may be useful for other use cases as well.

**Comment by antfitch** [comment]
Tentatively closing this due to the age and because it seems to have gone cold. I'm not sure where things are at today, but if this is still an issue, I'd open a new one with the what still needs to be fixed today.
